libs/sdk-core/src/boltzswap.rs

```diff
@@ -85,58 +84,163 @@ struct Pairs {
+
+    /// - Boltz: claim tx was seen, HODL invoice was settled
+    /// - Breez SDK: Marks the successful completion of a reverse swap
+    #[serde(rename = "invoice.settled")]
+    InvoiceSettled,
+
+    #[serde(rename = "invoice.expired")]
+    InvoiceExpired,
+
+    /// - Boltz: If Boltz is unable to send the agreed amount of onchain coins after the invoice is
+    /// paid, the status will become transaction.failed and the pending lightning HTLC will be cancelled.
+    #[serde(rename = "transaction.failed")]
+    LockTxFailed,
+
+    /// - Boltz: the HODL invoice was paid, but the timelock expired. In this case, the invoice expires
+    /// and the funds are returned to the sender.
+    #[serde(rename = "transaction.refunded")]
+    #[serde(rename_all = "camelCase")]
+    LockTxRefunded { failure_reason: String },
+
+    /// - Boltz: the timelock expires before the HODL invoice is paid
+    #[serde(rename = "swap.expired")]
+    SwapExpired,
+}
+
 pub struct BoltzApi {}
 
 #[tonic::async_trait]
 impl ReverseSwapperAPI for BoltzApi {
-    async fn reverse_swap_info(&self) -> Result<ReverseSwapPairInfo> {
-        crate::boltzswap::reverse_swap_info().await
+    async fn reverse_swap_pair_info(&self) -> Result<ReverseSwapPairInfo> {
+        reverse_swap_pair_info().await
     }
 
+    /// Call Boltz API and parse response as per https://docs.boltz.exchange/en/latest/api/#creating-reverse-swaps
+    ///
+    /// #### Errors
+    ///
+    /// This method returns an error for  HTTP or connection errors (404 not found, 400 bad request,
+    /// 502 server error, etc).
+    ///
+    /// Boltz API errors (e.g. if the reverse swap could not be created, for example if the amount is too low)
+    /// are returned as a successful response of type [BoltzApiCreateReverseSwapResponse::BoltzApiError]
     async fn create_reverse_swap(
         &self,
         amount_sat: u64,
-        onchain_claim_address: String,
+        preimage_hash_hex: String,
+        claim_pubkey: String,
         pair_hash: String,
         routing_node: String,
-    ) -> Result<ReverseSwap> {
-        let rand_bytes: [u8; 32] = random();
-        let preimage = sha256::Hash::hash(&rand_bytes);
-        let preimage_hash = sha256::Hash::hash(&preimage);
-        let preimage_hash_hex = preimage_hash.to_hex();
-
-        let response: CreateReverseSwapResponse = Client::new()
+    ) -> Result<BoltzApiCreateReverseSwapResponse> {
+        let response = Client::new()
             .post(CREATE_REVERSE_SWAP_ENDPOINT)
             .header(CONTENT_TYPE, "application/json")
             .body(get_boltz_reverse_swap_args(
                 amount_sat,
                 preimage_hash_hex,
-                pair_hash,
-                onchain_claim_address,
+                pair_hash.clone(),
+                claim_pubkey.clone(),
                 routing_node,
             ))
             .send()
             .await?
-            .json()
+            .text()
```

Comment:
Because `text()` lets us access the raw response body directly for debugging.

If I use `json()`, the automatic conversion to `BoltzApiReverseSwapStatus` kicks in and hides the raw json text.